src/silx/io/h5py_utils.py

```diff
@@ -64,6 +64,33 @@
 
 LATEST_LIBVER_IS_V108 = HDF5_HEX_VERSION < calc_hexversion(1, 10, 0)
 
+LOCKING_ARG_CHANGED_H5PY_VS_HDF5_WARMED = False
+
+
+def only_once(func):
+    """
+    simple decorator to make sure the warning is call at most one time.
+    """
+
+    def wrapper(*args, **kwargs):
+        global LOCKING_ARG_CHANGED_H5PY_VS_HDF5_WARMED
+        if not LOCKING_ARG_CHANGED_H5PY_VS_HDF5_WARMED:
+            LOCKING_ARG_CHANGED_H5PY_VS_HDF5_WARMED = True
+            return func(*args, **kwargs)
+
+    return wrapper
+
+
+@only_once
```

Comment:
You could (ab)use [functools.lru_cache](https://docs.python.org/3.8/library/functools.html#functools.lru_cache) here and save the global variable